pkg/storage/tsdb/config.go

```diff
@@ -90,3 +91,26 @@ func (cfg *Config) Validate() error {
 
 	return nil
 }
+
+// BucketStoreConfig holds the config information for Bucket Stores used by the querier
+type BucketStoreConfig struct {
+	Dir                  string `yaml:"dir"`
+	IndexCacheSizeBytes  uint64 `yaml:"index_cache_size_bytes"`
+	MaxChunkPoolBytes    uint64 `yaml:"max_chunk_pool_bytes"`
+	MaxSampleCount       uint64 `yaml:"max_sample_count"`
+	MaxConcurrent        int    `yaml:"max_concurrent"`
+	DebugLogging         bool   `yaml:"debug_logging"`
+	BlockSyncConcurrency int    `yaml:"block_sync_concurrency"`
+}
+
+// RegisterFlags registers the BucketStore flags
+func (cfg *BucketStoreConfig) RegisterFlags(f *flag.FlagSet) {
+
+	f.StringVar(&cfg.Dir, "experimental.tsdb.bucket-store-config.dir", "tsdb-sync", "directory to place synced tsdb indicies")
+	f.Uint64Var(&cfg.IndexCacheSizeBytes, "experimental.tsdb.bucket-store-config.index-cache-size", uint64(250*units.Mebibyte), "size of index cache in bytes")
+	f.Uint64Var(&cfg.MaxChunkPoolBytes, "experimental.tsdb.bucket-store-config.max-chunk-pool-bytes", uint64(2*units.Gibibyte), "max size of chunk pool in bytes")
+	f.Uint64Var(&cfg.MaxSampleCount, "experimental.tsdb.bucket-store-config.max-sample-count", 0, "max number of samples (0 is no limit)")
+	f.IntVar(&cfg.MaxConcurrent, "experimental.tsdb.bucket-store-config.max-concurrent", 20, "max number of concurrent go routines")
```

Comment:
I would improve the description. Isn't this the max number of concurrent queries executed against the storage? This is also per-tenant.